src/diffusers/pipelines/auto_pipeline.py

```diff
@@ -528,7 +528,12 @@ def from_pipe(cls, pipeline, **kwargs):
             if k not in text_2_image_kwargs
         }
 
-        missing_modules = set(expected_modules) - set(pipeline._optional_components) - set(text_2_image_kwargs.keys())
+        missing_modules = (
```

Comment:
I think maybe 
```
missing_modules = set(expected_modules) - set(text_2_image_cls._optional_components) - set(text_2_image_kwargs.keys())
```

Because if one component is optional for the other pipe (the pipe as in `.from_pipe(pipe)`) but it's not optional for the target pipe (text_2_image here) - it is still missing, no?

Just a thought: I didn't run any tests, I think it might a bug we had and was able to get away with this because the `_optional_componentes` is usually the same among different tasks for the same model group

so please let me know if it is the case:)

Reply:
Makes sense, thanks